network/codec/json/decode.go

```diff
@@ -6,192 +6,17 @@ import (
-		v = &messages.ChunkDataRequest{}
-	case CodeChunkDataResponse:
-		v = &messages.ChunkDataResponse{}
-
-	case CodeApprovalRequest:
-		v = &messages.ApprovalRequest{}
-	case CodeApprovalResponse:
-		v = &messages.ApprovalResponse{}
-
-	// generic entity exchange engines
-	case CodeEntityRequest:
-		v = &messages.EntityRequest{}
-	case CodeEntityResponse:
-		v = &messages.EntityResponse{}
-
-	// testing
-	case CodeEcho:
-		v = &message.TestMessage{}
-
-	// dkg
-	case CodeDKGMessage:
-		v = &messages.DKGMessage{}
-
-	default:
-		return nil, errors.Errorf("invalid message code (%d)", env.Code)
-	}
-
-	return v, nil
-}
-
-func switchenv2what(env Envelope) (string, error) {
-	var what string
-
-	switch env.Code {
-
-	// consensus
-	case CodeBlockProposal:
-		what = "CodeBlockProposal"
-	case CodeBlockVote:
-		what = "CodeBlockVote"
-
-	// cluster consensus
-	case CodeClusterBlockProposal:
-		what = "CodeClusterBlockProposal"
-	case CodeClusterBlockVote:
-		what = "CodeClusterBlockVote"
-	case CodeClusterBlockResponse:
-		what = "CodeClusterBlockResponse"
-
-	// protocol state sync
-	case CodeSyncRequest:
-		what = "CodeSyncRequest"
-	case CodeSyncResponse:
-		what = "CodeSyncResponse"
-	case CodeRangeRequest:
-		what = "CodeRangeRequest"
-	case CodeBatchRequest:
-		what = "CodeBatchRequest"
-	case CodeBlockResponse:
-		what = "CodeBlockResponse"
-
-	// collections, guarantees & transactions
-	case CodeCollectionGuarantee:
-		what = "CodeCollectionGuarantee"
-	case CodeTransactionBody:
-		what = "CodeTransactionBody"
-	case CodeTransaction:
-		what = "CodeTransaction"
-
-	// core messages for execution & verification
-	case CodeExecutionReceipt:
-		what = "CodeExecutionReceipt"
-	case CodeResultApproval:
-		what = "CodeResultApproval"
-
-	// execution state synchronization
-	case CodeExecutionStateSyncRequest:
-		what = "CodeExecutionStateSyncRequest"
-	case CodeExecutionStateDelta:
-		what = "CodeExecutionStateDelta"
-
-	// data exchange for execution of blocks
-	case CodeChunkDataRequest:
-		what = "CodeChunkDataRequest"
-	case CodeChunkDataResponse:
-		what = "CodeChunkDataResponse"
-
-	case CodeApprovalRequest:
-		what = "CodeApprovalRequest"
-	case CodeApprovalResponse:
-		what = "CodeApprovalResponse"
-
-	// generic entity exchange engines
-	case CodeEntityRequest:
-		what = "CodeEntityRequest"
-	case CodeEntityResponse:
-		what = "CodeEntityResponse"
-
-	// testing
-	case CodeEcho:
-		what = "CodeEcho"
-
-	// dkg
-	case CodeDKGMessage:
-		what = "CodeDKGMessage"
-
-	default:
-		return "", errors.Errorf("invalid message code (%d)", env.Code)
-	}
-
-	return what, nil
-}
-
 // decode will decode the envelope into an entity.
 func env2vDecode(env Envelope, via string) (interface{}, error) {
 
 	// create the desired message
-	v, err := switchenv2v(env)
-	if nil != err {
-		return nil, err
-	}
-
-	what, err := switchenv2what(env)
-	if nil != err {
-		return nil, err
+	v, what, err := codec.InterfaceFromMessageCode(env.Code)
```

Comment:
Since the JSON codec isn't being used as you mentioned, we can consider just removing it

Reply:
https://github.com/onflow/flow-go/pull/2491/commits/fdb5c242aa87a585612dcb2ff7fa549027ff17b7